plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java

```diff
@@ -57,13 +85,85 @@ public ConnectorSplitSource getSplits(
         HudiTableHandle hudiTable = (HudiTableHandle) tableHandle;
         SchemaTableName tableName = hudiTable.getSchemaTableName();
         HiveMetastore metastore = transactionManager.get(transaction).getMetastore();
-        Table table = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName())
+        Table table = metastore.getTable(identity, tableName.getSchemaName(), tableName.getTableName())
                 .orElseThrow(() -> new TableNotFoundException(tableName));
-        Configuration conf = hdfsEnvironment.getConfiguration(new HdfsEnvironment.HdfsContext(session), new Path(table.getStorage().getLocation()));
-        HudiSplitSource splitSource = new HudiSplitSource(hudiTable, conf);
+        HdfsEnvironment.HdfsContext context = new HdfsEnvironment.HdfsContext(session);
+        FileSystem fs = null;
+        try {
+            fs = hdfsEnvironment.getFileSystem(context, new Path(table.getStorage().getLocation()));
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+        }
+        Configuration conf = hdfsEnvironment.getConfiguration(context, new Path(table.getStorage().getLocation()));
+        HoodieTableMetaClient metaClient = hudiTable.getMetaClient().orElseGet(() -> getMetaClient(conf, hudiTable.getBasePath()));
+        List<String> partitionValues = getPartitionsWritten(metaClient.getActiveTimeline());
+        log.debug("Fetched partitions from Hudi: " + partitionValues);
+        hudiTable.getPartitions().ifPresent(p -> p.forEach(p1 -> log.debug("Partitions from TableHandle: " + p1)));
+
+        List<String> columnNames = table.getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+        log.debug("Column Names: " + columnNames);
+        HudiSplitSource splitSource;
+        String tablePath = table.getStorage().getLocation();
+        Optional<FileStatus[]> fileStatuses = Optional.empty();
+        if (!columnNames.isEmpty()) {
+            List<List<String>> partitionNames = metastore.getPartitionNamesByFilter(identity, tableName.getSchemaName(), tableName.getTableName(), columnNames, TupleDomain.all())
+                    .orElseThrow(() -> new TableNotFoundException(hudiTable.getSchemaTableName()))
+                    .stream()
+                    .map(HiveUtil::toPartitionValues)
+                    .collect(toImmutableList());
+            log.debug("Partition Names: " + partitionNames);
+
+            Optional<Partition> partition = metastore.getPartition(identity, table, partitionNames.get(0));
+
+            log.debug("Fetched partitions from Metastore: " + partition.get());
+            Properties schema = getPartitionSchema(table, partition);
+            String dataDir = schema.getProperty(META_TABLE_LOCATION);
+            log.debug("Partition schema: " + schema);
+
+            List<HivePartitionKey> partitionKeys = getPartitionKeys(table, partition);
```

Comment:
It looks like it only fetches the first partition.  Do we need to fix this?

Reply:
Yeah. In HudiSplitManager
```
List<List<String>> partitionNames = metastore.getPartitionNamesByFilter(identity, tableName.getSchemaName(), tableName.getTableName(), columnNames, TupleDomain.all())
```
This gives partition values for each partition column. We can create a map of partition column to value out of this.